source/sigv4.c

```diff
@@ -2653,10 +2655,10 @@ static SigV4Status_t writeLineToCanonicalRequest( const char * pLine,
         ( void ) memcpy( pCanonicalContext->pBufCur,
                          pLine,
                          lineLen );
-        pCanonicalContext->pBufCur += lineLen;
+        pCanonicalContext->pBufCur = &pCanonicalContext->pBufCur[ lineLen ];
 
         *( pCanonicalContext->pBufCur ) = LINEFEED_CHAR;
-        pCanonicalContext->pBufCur += 1U;
+        pCanonicalContext->pBufCur = &pCanonicalContext->pBufCur[ 1 ];
```

Comment:
```suggestion
        pCanonicalContext->pBufCur = &( pCanonicalContext->pBufCur[ 1 ] );
```

Reply:
Specify operator with parentheses in [this commit](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk/pull/92/commits/14e419b36f427f681805a2c987f0cdf37b2e0263).